### DISCUSSION TITLE
ppstructure-cpu版本运行时占用内存较高是否正常？

### DISCUSSION THREAD

**Discussion by IeohMingChan** [PaddlePaddle/PaddleOCR#14204] [Q&A] [answered]
我使用ppstructure处理一个35页约850KB的pdf文件时占用物理内存打到10.3g，虚拟内存达到17g，请问这种现象是否正常，且是否有降低占用内存的方法？我的版本是paddleocr==2.8.1, paddlepallde==2.6.2。感激不尽！

**Comment by GreatV** [comment]
@IeohMingChan 可以提供一个最小可复现的demo吗，处理代码和对应的pdf文件。

**Reply by IeohMingChan** [reply]
`import io
import json
import uuid
import re
import time
from fastapi import FastAPI, Form, BackgroundTasks, APIRouter
from fastapi.responses import JSONResponse
from io import BytesIO
from paddleocr import PPStructure
from loguru import logger
from ocr_config import *
app = FastAPI()

from typing import List, Dict
import pandas as pd
from bs4 import BeautifulSoup
import httpx
class OCRService:
def init(self):

    self.return_table = True
    self.CATALOG_DETECTION_THRESHOLD = 3
    self.ocr_engine = PPStructure(table=self.return_table, ocr=True, show_log=False, structure_version="PP-StructureV2")

def process_file(self, file_path: str):

    result = self.ocr_engine(file_path)
    return result
ocr_service = OCRService()

@app.post("/ocr")
def ocr(file_path: str = Form(...), isMinio: bool = Form(...)):
try:

    result = ocr_service.process_file(file_path, isMinio)
    return JSONResponse(content={"status": "success", "dict_list": result}, status_code=200)


except Exception as e:
    logger.error(f"OCR处理失败: {e}")
    return JSONResponse(content={"status": "error", "message": str(e)}, status_code=500)
if name == "main":
import uvicorn
uvicorn.run('ocr_service:app', host="0.0.0.0", port=1300, workers = 1)`

[AIprompt52.pdf](https://github.com/user-attachments/files/18681509/AIprompt52.pdf)

补充：目前已将paddlepaddle升级至3.0.0rc1，paddleocr版本不变，但问题仍然存在

**Comment by myhloli** [comment, marked as answer]
paddlepaddle 2.6.x适合使用的版本为paddleocr 2.7.3，
paddleocr2.8以上的版本需要使用paddlepaddle 3.0
虽然我不知道原因，但是就是这么离谱

**Reply by IeohMingChan** [reply]
paddlepaddle升级成3.0版本后问题仍然存在
